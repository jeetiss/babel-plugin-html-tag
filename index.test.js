'use strict';

const babel = require('@babel/core');
const path = require('path');
const { readFileSync } = require('fs');

const plugin = require('./index');

function transform(
  code,
  opts = {
    tags: ['html'],
    sortAttributes: true,
    sortClassName: true,
    collapseWhitespace: true,
    conservativeCollapse: false,
    removeComments: true,
    removeEmptyAttributes: true,
    removeTagWhitespace: true,
  },
) {
  return babel.transform(code, {
    plugins: [[plugin, opts]],
  }).code;
}

describe('babel-plugin-html-tag', () => {
  test('returns tagged template', () => {
    const res = transform('html`<p>\n${ textToTest }\n</p >`');
    console.log(res);
    expect(res).toMatchSnapshot();
  });

  test('returns static string when possible', () => {
    const res = transform('html`<p class="center">AAA bbb, \'fff\'</p>`');
    console.log(res);
    expect(res).toMatchSnapshot();
  });

  test('fixture test', () => {
    const code = readFileSync(
      path.resolve(__dirname, './__fixtures__/a.js'),
      'utf8',
    );
    expect(transform(code)).toMatchSnapshot();
  });
});
