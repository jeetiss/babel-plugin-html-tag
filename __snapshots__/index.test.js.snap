// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-html-tag fixture test 1`] = `
"'use strict';
/* global html */

const b = 'test';
const a = \`<p class=\\"center justify zoom\\">This is paragraph with \${b} subsitutions at several lines: \${1 + 2}</p>\`;
const z = '<table class=\\"center\\"><tr class=\\"left\\"><td>HTML without substitutions</td></tr></table>';"
`;

exports[`babel-plugin-html-tag returns static string when possible 1`] = `"'<p class=\\"center\\">AAA bbb,\\\\'ff\\\\'</p>';"`;

exports[`babel-plugin-html-tag returns tagged template 1`] = `"\`<p>\${textToTest}</p>\`;"`;
